### PR TITLE
Gunstore overhaul

### PR DESCRIPTION
**Pull request by Invalidated-User**
#### Summary

Content "overhauling gun stores"

#### Purpose of change

Makes gun stores more dynamic and believable

#### Describe the solution

Created gunstore_palette and applied that palette to all existing gun store mapgens.

Created dynamic looted versions of all gun store variants that did not have one already by creating and utilizing nested mapgens

All ammunition was moved to be inside gun store safes

Altered gun store item spawns by creating the SUS_gunstore item category and increasing the spawn rates across the board.

Removed many weapons which are illegal to sell from the gun store display category as well as some minor corrections of the ammo, magazine, and mod item groups.

Added rare (15%) copbot spawns to unlooted gun stores to justify ar-15s and other weapons being in stock which are illegal for purchase within MA. It is reasonable to consider that the state governments of New England (specifically MA) in the cata timeline would have considered deploying autonomous copbots to the point of sale at gun stores as a novel alternative to reinstating the defunct AWB as they did in 2016 of the OTL. ( https://www.mass.gov/guides/frequently-asked-questions-about-the-assault-weapons-ban-enforcement-notice for an easy read on the topic) Creating a believable pool of items for sale with that law being in effect would be incredibly difficult

Fixed minor spelling mistake from #50436

#### Describe alternatives you've considered

Not applying the new gunstore_palette to existing mapgens as well as removing the copbots.

dividing this PR and submitting them as many separate requests.

#### Testing

Spawned and visited all gun store variants as well as found organically spawned versions of all variants. Inspected all item spawn locations to ensure they were working correctly

#### Additional context

Looted gunsmith:
Cataclysm-vcpkg-static-Release-x64_fJxSo2VTNA
Regular gunsmith:
Cataclysm-vcpkg-static-Release-x64_KxRxburbk8

### CODE REVIEW

File: data/json/itemgroups/Weapons_Mods_Ammo/guns.json [hunk 1]
```diff
@@ -183,6 +208,18 @@
...
   },
+  {
+    "type": "item_group",
+    "subtype": "distribution",
+    "id": "guns_pistol_obscure_display",
+    "//": "Very obscure pistols which are illegal to sell.",
+    "items": [
+      { "item": "af2011a1_38super", "prob": 25 },
+      { "item": "draco", "prob": 30, "charges-min": 0, "charges-max": 0 },
+      { "item": "ar_pistol", "prob": 35, "charges-min": 0, "charges-max": 0 },
```
Comment (LaVeyanFiend) [inline data/json/itemgroups/Weapons_Mods_Ammo/guns.json:219]: None of these are illegal to sell 🤔

**Comment by Maleclypse** [comment]
Content "overhauling gun stores" will fix the PR validator test.

**Comment by Night-Pryanik** [comment]
It seems like there's way too many guns lying on the floor after the looting. I'd more believe in at most 1-2 guns.

**Comment by Pseudogeist** [comment]
What's the rationale behind ammo in gun safes? Is this only in looted stores? Having looked at a few New England gun stores online there's plenty of ammo on display behind the counter, which is what I believed to be standard practice. [Old Glory Guns & Ammo, Greenville, NH](https://www.oldglorygunsandammo.com/wp-content/uploads/2015/08/image41.jpeg). Besides that, the sheer amount of guns is _staggering_. Also, Old Glory deal in so-called Class III firearms, which they describe [here](https://www.oldglorygunsandammo.com/nfa-class-iii-weapons/) so there's definitely precedent for selling automatics and such. How common I couldn't say, that would require some research. I'm inclined to think Old Glory would rather you not bother shopping around so I'm not going to trust them on their claim of being the 'premier' Class III dealer for New Hampshire, but that does at least imply there are others.

**Comment by Invalidated-User** [comment]
> What's the rationale behind ammo in gun safes? Is this only in looted stores? Having looked at a few New England gun stores online there's plenty of ammo on display behind the counter, which is what I believed to be standard practice. [Old Glory Guns & Ammo, Greenville, NH](https://www.oldglorygunsandammo.com/wp-content/uploads/2015/08/image41.jpeg). Besides that, the sheer amount of guns is _staggering_. Also, Old Glory deal in so-called Class III firearms, which they describe [here](https://www.oldglorygunsandammo.com/nfa-class-iii-weapons/) so there's definitely precedent for selling automatics and such. How common I couldn't say, that would require some research. I'm inclined to think Old Glory would rather you not bother shopping around so I'm not going to trust them on their claim of being the 'premier' Class III dealer for New Hampshire, but that does at least imply there are others.

I put all the ammunition spawns in gunsafes as a way of representing how highly demanded and valued they would've been in the days leading up to the cataclysm. Great point about NFA Class III firearms being available for transfer from a store to an individual! I wasn't sure how common that practice was so I had created a separate "SUS_gunstore_safe_obscure" category which only spawned in 2 out of the 5 unlooted gunstore variants' BoH in addition to the normal SUS_gunstore_safe which would appear in the FoH (not the looted versions atm, but I could replace the normal SUS_gunstore_safe pool in those if viewed as appropriate)

**Comment by Pseudogeist** [comment]
> 
> I put all the ammunition spawns in gunsafes as a way of representing how highly demanded and valued they would've been in the days leading up to the cataclysm. 

Hmm... Fair enough. I suppose the variables are the size of safe, availability of the stock and the mindset of the store owner. If ammo stocks were low and demand super high I can see that being the case. That said, I know nothing about gun safes in gun stores. I'd have thought the store itself would have to be something of a safe, or the stock room would be more of a vault. I would have thought any safe would be used solely for storing money and weapons that (for whatever reason) couldn't be displayed. My knowledge is minimal at best, though. I'd be hesitant about adding more/larger gun safes than average (whatever _that_ is) as I imagine procurement and installation isn't a quick thing at the best of times, but that's pure speculation. I seriously doubt there's many <s>gun</s> stores who advertise the specifications of their safes on their website so I'm not sure how best to research that.

>
>Great point about NFA Class III firearms being available for transfer from a store to an individual! I wasn't sure how common that practice was so I had created a separate "SUS_gunstore_safe_obscure"(Containing NFA violating or very obscure firearms) category which only spawned in 2 out of the 5 unlooted gunstore variants' BoH in addition to the normal SUS_gunstore_safe which would appear in the FoH (not the looted versions atm, but I could replace the normal SUS_gunstore_safe pool in those if viewed as appropriate)

Glad to be of service! I wasn't sure whether to say anything or not as I have no experience or opinion on the matter, but that material looked potentially useful.

**Comment by jorganmir** [comment]
You haven't considered pre-ban AR-15's, which can be bought in Massachusetts, or pre-ban lower receivers, which can also be used to make legal AR-15's in MA.

https://www.thegunparlor.com/pages/preban-firearms

https://tombstonetrading.com/firearms/used-gun-sales/

While expensive, they certainly exist, and shouldn't be entirely ruled out. Additionally, unless 80% lowers and other means of building an AR are accounted for mechanically, it doesn't quite make sense to throttle player supply of AR rifles too heavily.

https://www.thegunparlor.com/pages/rifles-in-store?subcategory_id=163909&caliber_id=289%2C127%2C158%2C179%2C360%2C153%2C306%2C319%2C382%2C322&select_out_of_stock=&local=true&page=1

Using this search of semiautomatic rifles in rifle chamberings only, notice that this search only looks at inventory held in-store, and it is reasonable that usable AR-15's should remain in gun store inventories. Keeping the rifles in a safe rather than on shelves is the most reasonable approach.

**Comment by Zireael07** [comment]
@jorganmir: Note that the game is set in New England, not Massachusetts.

**Comment by jorganmir** [comment]
> @jorganmir: Note that the game is set in New England, not Massachusetts.

https://en.wikipedia.org/wiki/Gun_laws_in_New_Hampshire
https://en.wikipedia.org/wiki/Gun_laws_in_Vermont
https://en.wikipedia.org/wiki/Gun_laws_in_Maine
https://en.wikipedia.org/wiki/Gun_laws_in_Rhode_Island
https://en.wikipedia.org/wiki/Gun_laws_in_Connecticut
https://worldoftroy.com/shop-now/    (Connecticut legal AR rifles)
https://www.reddit.com/r/ar15/comments/k6aama/connecticut_legal_ar_porn/   (Connecticut legal AR's with normal magazines) 

Gun laws have many oddities and exceptions to them. One can own an AR in more or less every state traditionally held to be in New England, either through AOW "pistol" AR's (which are probably below resolution of the game to stat significantly differently.), that skirt by those laws, grandfathering, components, or other means. It isn't really that well supported to remove "assault weapons" from gun stores, and law and lawsuits can change by the time the apocalypse occurs.

**Comment by tenmillimaster** [comment]
Gun stores would just close up shop if AWB's were enforced, not run with cop bots inside. Who would they even be selling to at that point?

Many FFL's already have to navigate different types of sales non-NFA and NFA items, which the time scales for transfer differ wildly for. They already do so without copbot supervision.

**Comment by Invalidated-User** [comment]
> Gun stores would just close up shop if AWB's were enforced, not run with cop bots inside. Who would they even be selling to at that point?

Evidently not in MA where "Attorney General Maura Healey issued a notice on July 20, 2016 to all gun sellers and manufacturers in Massachusetts, warning that her office is stepping up enforcement of the state’s assault weapons ban, including a crackdown on the sale of copycat weapons." https://www.mass.gov/enforcing-the-massachusetts-assault-weapons-ban

I do not believe that is ethical or legal to deploy copbots at the point of sale at gunstores to enforce gunlaws, but neither is the deployment of turrets to suppress rioters and the deployment of CROWS turrets in prisons. I think the previous 2 examples are representative of a general trend in CATA of state governments preferring to utilize novel (and potentially immoral/illegal) technological solutions to social problems over legislative ones. 

With that in mind, my intent behind the inclusion of rare (15%) copbot spawns in unlooted stores is to provide a justification behind the state governments of New England (mostly MA) preferring to employ a technological solution to the issue of gun control over the legislative (enforcing the AWB) ones employed in the OTL.

I am not a legal scholar or an expert in gun control regulations and thus a 100% accurate distribution of guns within the spawn pools is beyond the scope of this PR. My intent was to create a uniform set of spawn pools through the creation of the gun store palette and SUS_group.

With all that in mind, I am ready to roll back my inclusion of cop bot spawns and my alterations to the gun store spawn pools if the general consensus is that they should be reverted.

On a different note, this PR has been up for nearly 3 weeks now without a full review, is there something wrong with my PR's format or code? I know that the Github has been incredibly busy since the recent release of 0.F-1, but I am fairly inexperienced so it would make sense if I have made some sort of mistake that has prevented it from being reviewed.

**Review by curstwist — COMMENTED** [review]
The mapgen portions of this look good, so I can approve that portion.  

I think others have weighed in on the gun item spawns & the cop bots and I don't have anything else to add for those.

**Comment by tenmillimaster** [comment]
> Evidently not in MA where "Attorney General Maura Healey issued a notice on July 20, 2016 to all gun sellers and manufacturers in Massachusetts, warning that her office is stepping up enforcement of the state’s assault weapons ban, including a crackdown on the sale of copycat weapons." https://www.mass.gov/enforcing-the-massachusetts-assault-weapons-ban

Ok, but it would be easier to just offload your inventory online to non-MA residents on GunBroker or something rather than operate with a copbot in your storefront... for some reason? 

The governments of CDDA are not radically different from ours, save for a brief period of a few months. During that time, it's likely copbots could be better used elsewhere than babysitting an FFL.

**Comment by Invalidated-User** [comment]
> > Evidently not in MA where "Attorney General Maura Healey issued a notice on July 20, 2016 to all gun sellers and manufacturers in Massachusetts, warning that her office is stepping up enforcement of the state’s assault weapons ban, including a crackdown on the sale of copycat weapons." https://www.mass.gov/enforcing-the-massachusetts-assault-weapons-ban
> 
> Ok, but it would be easier to just offload your inventory online to non-MA residents on GunBroker or something rather than operate with a copbot in your storefront... for some reason?
> 
> The governments of CDDA are not radically different from ours, save for a brief period of a few months. During that time, it's likely copbots could be better used elsewhere than babysitting an FFL.

I understand that the copbot solution to the MA problem is not a 100% foolproof one, but your line of reasoning does not address the underlying problem that it was created to solve. Unless you have some alternative solution to MA's recent enforcement of the AWB I haven't thought of an alternate method for the creation of a unified gunstore firearms loot pool. Otherwise, someone who was an expert in this topic would have to create 6 separate loot pools for each individual state to represent the divergent gun laws of NE

This whole conversation is representative of why I wanted to add copbots to gun stores in the first place: it creates a "deus ex machina" solution to what is an incredibly controversial and contentious issue that is well beyond the scope of being accurately modeled within CDDA.

**Comment by faefux** [comment]
Hey @Invalidated-User, what's going on with this?  There seems to be a conflict.

**Comment by I-am-Erk** [comment]
I'd like to see this get finished and merged. @Invalidated-User are you still around or should we just polish off any outstanding things that arise

**Comment by jbytheway** [comment]
@Invalidated-User In the `s_gunstore` mapgen I see that you made `;` be `t_gutter_downspout` in the palette, but then it's overriden to be `t_region_groundcover_urban` in both the places that use the palette.  I suspect that's a mistake and it was supposed to be `t_gutter_downspout`.  I'm changing it to that (i.e. removing the `t_region_groundcover_urban`); let me know if I misunderstood your intent.

(This is as part of a batch of assorted mapgen fixes I'm working on.  PR to come at some point in the future)

**Pull request merged**